projects/sandbox/deploy/deploy/trigger.py

```diff
@@ -0,0 +1,158 @@
+import json
+import logging
+import time
+from dataclasses import asdict, dataclass
+from pathlib import Path
+from typing import List, Literal, Optional, Union
+
+import numpy as np
+from ligo.gracedb.rest import GraceDb
+
+from bbhnet.analysis.ledger.events import TimeSlideEventSet
+
+Gdb = Literal["playground", "test", "production"]
+
+
+@dataclass
+class Event:
+    time: float
+    detection_statistic: float
+    far: float
+
+
+class Searcher:
+    def __init__(
+        self,
+        outdir: Path,
+        far_per_day: float,
+        inference_sampling_rate: float,
+        refractory_period: float,
+        offset: float,
+    ) -> None:
+        logging.debug("Loading background measurements")
+        background_file = outdir / "infer" / "background.h5"
+        self.background = TimeSlideEventSet.read(background_file)
+
+        num_events = int(far_per_day * self.background.Tb / 3600 / 24)
+        if not num_events:
+            raise ValueError(
+                "Background livetime {}s not enough to detect "
+                "events with false alarm rate of {} per day".format(
+                    self.background.Tb, far_per_day
+                )
+            )
+
+        events = np.sort(self.background.detection_statistic)
+        self.threshold = events[-num_events]
+        logging.info(
+            "FAR {}/day threshold is {:0.3f}".format(
+                far_per_day, self.threshold
+            )
+        )
+
+        self.inference_sampling_rate = inference_sampling_rate
+        self.refractory_period = refractory_period
+        self.offset = offset
+        self.last_detection_time = time.time()
+        self.detecting = False
+
+    def check_refractory(self, value):
+        time_since_last = time.time() - self.last_detection_time
+        if time_since_last < self.refractory_period:
+            logging.warning(
+                "Detected event with detection statistic {:0.3f} "
+                "but it has only been {}s since last detection, "
+                "so skipping".format(value, time_since_last)
+            )
+            return True
+        return False
+
+    def build_event(self, value: float, t0: float, idx: int):
+        if self.check_refractory(value):
+            return None
+
+        timestamp = t0 + self.inference_sampling_rate * idx
+        timestamp += self.offset
+
+        far = self.background.far(value)
+        far /= 365 * 24 * 3600
```

Comment:
Just to be picky here: the `.far()` converts from seconds to years using 31556952 seconds per year

Reply:
Good call, mean to look this up but was too lazy so thank you for providing that